### PR TITLE
Add service, connector and package init functions

### PR DESCRIPTION
**Pull request by anupama-pathirage**

### CODE REVIEW

File: modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/SymbolEnv.java [hunk 1]
```diff
@@ -84,6 +84,14 @@ public static SymbolEnv createPkgLevelSymbolEnv(BLangNode node,
         return symbolEnv;
     }
 
+    public static SymbolEnv createSubLevelSymbolEnv(BLangNode node,
```
Comment (sameera-jayasoma) [inline modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/SymbolEnv.java]: Shall we rename this to "createResourceActionSymbolEnv"?

File: modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java [hunk 2]
```diff
@@ -326,7 +339,12 @@ private void defineInvokableSymbol(BLangInvokableNode invokableNode, BInvokableS
     }
 
     private void defineInvokableSymbolParams(BLangInvokableNode invokableNode, BInvokableSymbol symbol) {
-        SymbolEnv invokableEnv = SymbolEnv.createPkgLevelSymbolEnv(invokableNode, env, symbol.scope);
+        SymbolEnv invokableEnv;
+        if (invokableNode instanceof BLangAction || invokableNode instanceof BLangResource) {
```
Comment (sameera-jayasoma) [inline modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java]: Shall we pass the env as a parameter to this method? The correct env can be created in the following methods.

public void visit(BLangFunction funcNode) 
public void visit(BLangAction actionNode)
public void visit(BLangResource resourceNode)

File: modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java [hunk 3]
```diff
@@ -359,4 +377,101 @@ public void defineNode(BLangNode node, SymbolEnv env) {
         node.accept(this);
         this.env = prevEnv;
     }
+
+    private void defineConnectorInitFunction(BLangConnector connector) {
```
Comment (sameera-jayasoma) [inline modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java]: Looks good. 

But seems like we have common logic repeated in all three methods. Can we refactor them?

File: modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java [hunk 4]
```diff
@@ -76,6 +85,7 @@
 
     private SymbolEnv env;
     public Map<BPackageSymbol, SymbolEnv> packageEnvs = new HashMap<>();
+    public static final String INIT_FUNCTION_SUFFIX = ".<init>";
```
[the remark this answers is not in the document]
Comment (anupama-pathirage) [reply modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java]: Fixed with 6a6642b8094d731de9e332c0f6b3a52a7b54bfcc

File: modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java (hunk 2 above)
Comment (anupama-pathirage) [reply modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java]: Fixed with 6a6642b8094d731de9e332c0f6b3a52a7b54bfcc

File: modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java (hunk 3 above)
Comment (anupama-pathirage) [reply modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java]: Fixed with 6a6642b8094d731de9e332c0f6b3a52a7b54bfcc

**Pull request merged**